### ISSUE TITLE
folder structure view of library

### ISSUE THREAD

**Issue by felixcode** [andrewrk/groovebasin#93]
current view of library is not very suitable for library that music tags/info isn't very good.
so maybe it's better to also provide a folder structure view from which we can choose files to play directly.

**Comment by MartNytrm** [comment]
Hey guys and @andrewrk, 
would love to see this feature, too, because going to your library and finding things is pretty damn hard. You only can use this little: 

![selection_010](https://cloud.githubusercontent.com/assets/11982465/7775887/388f7538-00b7-11e5-8cb2-85e0b02446e4.png)

Features I vote for:
- folder view (this is the way still many people organize their music)
- and better sorting view for library (show albums first, ...)

Thanks for Groove Basin! It's pretty unique!!

Greets


Nytrm

**Comment by andrewrk** [comment]
Album view is #170

**Comment by nubjub** [comment]
I'm thinking about switching my player system over to this, but folder / file view is what i have to have before I can do this. Do you have any insights on how this should best be implemented so that I can have a look at getting to work on it?

**Comment by andrewrk** [comment]
Yes. Much of the work is already done via the mpd protocol support. The server has an in memory index of the current file structure that it keeps up to date. What's left to be done is communicating this information to the client and updating the client code to use this information. 

When I get home I'll elaborate on the details. You can also find me in the #libgroove freenode irc channel for live support.

**Comment by andrewrk** [comment]
In lib/player.js look at `dbFilesByPath` as well as `dirs`. These objects are kept up to date and contain the file system hierarchy.

In lib/player_server.js you will have to add a new subscription or modify an existing one to communicate the file system hierarchy to the client. Also take a look at the [Protocol Specification](https://github.com/andrewrk/groovebasin/blob/master/doc/protocol.md) for inspiration - whatever mechanism you add will have to be added to the protocol spec as well.

Once you have the server communicating data to the client, you'll have to edit src/client/app.js and probably src/public/index.html.

I hope that helps. This feature is a big undertaking; don't feel bad if it's too hard or you don't have time. I'm happy to help and answer any questions you have.
